Fix cache push for spack compilers

douglasjacobsen opened the pull request:
Recently, the compiler_specs_for_environment method was updated to separate the package and the compiler spec. The spack package manager was not updated correctly for this.

This merge fixes this issue.

douglasjacobsen committed:
Fix cache push for spack compilers

Recently, the compiler_specs_for_environment method was updated to
separate the package and the compiler spec. The spack package manager
was not updated correctly for this.

This commit fixes this issue.

linsword13 commented on var/ramble/repos/builtin/package_managers/spack-lightweight/package_manager.py:
```diff
@@ -378,9 +378,11 @@ def _push_to_spack_cache(self, workspace, app_inst=None):
                 app_context, app_inst.expander, self, require=require_env
             )
-            compiler_specs = software_envs.compiler_specs_for_environment(
+            compiler_specs = []
```
nit: may be a slightly more concise version:

```
compiler_specs = [
    spec for spec, _ in software_envs.compiler_specs_for_environment(software_env)
]
```

douglasjacobsen committed:
Add tests of spack mirror and push-to-cache

This commit adds tests to exercise spack's mirror and push-to-cache
pipelines. These don't actually perform the spack actions, and instead
simply exercise Ramble's side of these pipelines.

douglasjacobsen committed:
Update bash completion

douglasjacobsen committed:
Fix invalid error reference in spack
